src/vs/workbench/parts/debug/electron-browser/debugService.ts

```diff
@@ -1064,4 +1065,27 @@ export class DebugService implements IDebugService {
 			error: message
 		});
 	}
+
+	private telemetryDebugAddBreakpoint(breakpoint: IBreakpoint, context: string): TPromise<any> {
+		/* __GDPR__
+			"debugAddBreakpoint" : {
+				"type" : { "classification": "SystemMetaData", "purpose": "FeatureInsight" },
+				"context": { "classification": "SystemMetaData", "purpose": "FeatureInsight" }
+			}
+		*/
+
+		var breakpointType = 'Breakpoint';
```

Comment:
Use `let` not `var`

Reply:
I do not like that we have this logic here: that we determing the type fo the breakpoint based on its characterstics. Do you really need this?
If yes I suggest to move this to be a field in the breakpoint called typeLabel and it shuold also be used in all places where we do this, for example [here](https://github.com/Microsoft/vscode/blob/6ba3f80ff402291d87e358624044d401a52e3547/src/vs/workbench/parts/debug/browser/breakpointsView.ts#L586)

Reply:
Yes we need this to be able to get an overview of the breakpoint types. Alternatively we can send the properties like `hasCondition`, `hasHitCondition` and `hasLogMessage` and do the aggregation as a part of the query? That way we move the logic our of VS Code core, and can easily re-categorize?

I wasn't aware of that functionality in the `breakpointsView` which really is render specific, but conceptually the same logic.